### ISSUE TITLE
Failed to deserialize `Signature`

### ISSUE THREAD

**Issue by fastpok** [RustCrypto/elliptic-curves#608]
Hi! I've encountered an error when deserializing a `Signature` from a JSON value.

Here's my test:
```rust
use serde::{Deserialize, Serialize};
use p256::ecdsa::{signature::Signer, Signature, SigningKey};
use rand_core::OsRng;

#[test]
fn test_signature_serialization() {
    let signing_key = SigningKey::random(&mut OsRng);
    let message = b"test";
    let signature = signing_key.sign(message);
    let serialized_signature = serde_json::to_value(&signature).unwrap();
    let deserialized_signature: Signature = serde_json::from_value(serialized_signature).unwrap();
    assert_eq!(signature, deserialized_signature);
}
```

Dependencies:
```toml
[dependencies]
p256 = { version = "0.11.1", features = ["serde"] }
serde = { version = "1.0", features = [ "derive" ] }
serde_json = "1.0"
rand_core = { version = "0.6.3", features = ["std"] }
```

Result:
```
thread 'state::tests::test_signature_serialization' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: string \"AE352B160759AACCA10953EF0F84B740A635AB39A86D6935DB6437B70CA4F6A943CEB4FF51562B00B9E3444E6914589839E96A14EDE4E297C0187856A6A9DCF8\", expected a borrowed string", line: 0, column: 0)'
```

**Comment by tarcieri** [comment]
Looks like we never migrated the `ecdsa` crate to use the new `serde` support in the `elliptic-curve` crate.

Thanks for the report.

**Comment by tarcieri** [comment]
Fixed in `ecdsa` v0.14.2

**Comment by fastpok** [comment]
Wow, that was fast, thanks!
